orderer/common/filerepo/filerepo_test.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package filerepo_test
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/hyperledger/fabric-protos-go/common"
+	"github.com/hyperledger/fabric/orderer/common/filerepo"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestFileRepo_Save(t *testing.T) {
+	options := filerepo.Options{
+		FileSuffix:          ".joinblock",
+		TransientFilePrefix: "~",
+		FileRepoDir:         "testdata",
+	}
+
+	r := filerepo.Repo{
+		Options: options,
+	}
+
+	t.Run("Successful save", func(t *testing.T) {
+		err := r.Save("newchannel", []byte("block-bytes"))
+		defer os.Remove("testdata/newchannel.joinblock")
+		assert.NoError(t, err)
+
+		// Check file bytes
+		bytes, err := ioutil.ReadFile("testdata/newchannel.joinblock")
+		require.NoError(t, err)
+		assert.Equal(t, bytes, []byte("block-bytes"))
+
+		// Check tempfile doesn't exist
+		_, err = os.Stat("testdata/~newchannel")
+		assert.True(t, os.IsNotExist(err))
+	})
+
+	t.Run("Failed save, invalid channel name", func(t *testing.T) {
+		err := r.Save("InvalidChannelName", []byte{})
+		assert.EqualError(t, err, "'InvalidChannelName' contains illegal characters")
+	})
+}
+
+func TestFileRepo_Scan(t *testing.T) {
+	options := filerepo.Options{
+		FileSuffix:          ".joinblock",
+		TransientFilePrefix: "~",
+		FileRepoDir:         "testdata",
+	}
+
+	r := filerepo.Repo{
+		Options: options,
+	}
+
+	t.Run("Successful scan", func(t *testing.T) {
+		files, err := r.Scan()
+		assert.NoError(t, err)
+		assert.Equal(t, files, []string{"mychannel.joinblock"})
```

Comment:
Whenever testing a function with a loop, I would always prefer to have at least two iterations.

Reply:
I believe it already iterates twice, once to add `mychannel.joinblock` and the other to filter out `mychannel.block` which doesn't match the file suffix pattern.